prometheus_client/registry.py

```diff
@@ -135,10 +135,23 @@ def __init__(self, names, registry):
         self._registry = registry
 
     def collect(self):
-        for metric in self._registry.collect():
-            m = metric._restricted_metric(self._name_set)
-            if m:
-                yield m
+        names = copy.copy(self._name_set)
+        collectors = set()
+        yield_target_info = False
+        with self._registry._lock:
+            if 'target_info' in names and self._registry._target_info:
+                yield_target_info = True
+                names.remove('target_info')
```

Comment:
You don't need this, the registry should be preventing this from happening in the first place.

Reply:
Not sure I understood it right - but done.